README.md

```diff
@@ -1,28 +1,133 @@
-# inmemory-pagination
+# In Memory-POSF
 
-## Filter & Search :
 
-Filtering is done by column name while search is performed across columns.
-Multiple values can be supplied at the same time.
+### Problem Statement
 
-`&` -> separates column names
-`|` -> separates multiple values
-`*` -> indicates search across all columns
+Struck with Datasource with no filtering,searching and pagination features?.
+UI is feeling Hard to handle bulk of data?.
+We provide you the best solution to overcome these issues with minimal overhead.
 
-### Examples:
+### Solution
 
-"name=bharath"
-"*=Vega|vegas"
-"multiMedia.[].name=CCCC&*=Vega|vegas&userId=4051"
+As we are developing so much software in today’s world and dealing with tons of data.
+Imagine a situation where you have tons of data to show on your web page.
+A browser will have not able to render all the data at one shot.
+So the best option is to go for InMemory pagination, where a server will return the bounded number of rows to show in the UI.
 
-## TODO : 
 
-1. write logic for filtering
-2. write logic for searching
-3. add enough log statement
-4. extend logic to support nested objects ( filtering, search and sort)
-5. create excception class
-6. Throw exception column name is invalid(if column name is not exist)
-7. Write unit testcases (use cucumber)
-8. document on how to use with example
-9. Finally publish to maven central repo
+### How to use this plugin
+
+##### Approach 1
+
+- To use this plugin , just extend the class **PaginationService** and override the **getRawData()**  function such that it returns list of desired object.
+- *getRawData()* method will return the complete source data from which the plugin will take care of performing pagination and searching for you.
+
+Sample snippet:
+
+https://github.com/techconative/inmemory-posf/blob/536ba7b1062edce986798a96c7c7210302921807/src/test/java/com/techconative/inmemory/pagination/FeedService.java#L23-L34
+
+<br> <br>
+
+##### Approach 2
+
+- Directly call static methods of class InmemoryFOPS Class that the plugin provides by default.
+- use *applyFiltering*  which is static method in InmemoryFOPS for filtering and searching.
+- it takes List of Map and PaginationCriteria as parameters.
+- Can be used anywhere in your program depending on your use cases.
+
+Sample usage:
+
+```java
+  InmemoryFOPS.applyFiltering( PaginationCriteria criteria, List<Map<String, String>> rawData)
+```
+ 
+- Same way ,we can call *applySorting* and *applyPagination* static method of InmemoryFOPS to leverage its benefits.
```

Comment:
I don't believe so. Those methods are private and should not be accessible from outside the lib

Reply:
i didn't see private .. my bad .. 

have changed it now